### PR TITLE
[docs] Spruce up experimental/legacy tags in Sphinx-generated docs

### PR DESCRIPTION
**Pull request by hellendag**
## Summary & Motivation

Use "Experimental" and "Legacy" tag style from elsewhere in Docs within Sphinx-generated API docs.

Screenshot 2023-09-12 at 4 52 45 PM
Screenshot 2023-09-12 at 4 53 16 PM
Screenshot 2023-09-12 at 4 53 51 PM

## How I Tested These Changes

Build docs, view site. Verify that tags appear.

### CODE REVIEW

**Comment by hellendag** [comment]
Current dependencies on/for this PR:
* master
  * **PR #16467** <a href="https://app.graphite.dev/github/pr/dagster-io/dagster/16467" target="_blank">Graphite</a>  👈

This comment was auto-generated by [Graphite](https://app.graphite.dev/github/pr/dagster-io/dagster/16467?utm_source=stack-comment).

File: docs/next/styles/globals.css [hunk 1]
```diff
@@ -99,6 +99,16 @@ dt > a.reference.internal {
   @apply absolute top-3 right-3 opacity-50 text-sm;
 }
 
+.experimental-tag,
+.flag.experimental {
+  @apply inline-flex items-center px-3 py-0.5 rounded-full align-middle text-xs uppercase font-medium bg-sea-foam text-gable-green
```
Comment (hellendag) [inline docs/next/styles/globals.css:104]: This combines Tailwind classes together to be applied to the `.experimental-tag` and `.flag.experimental` selectors.

File: docs/sphinx/_ext/dagster-sphinx/dagster_sphinx/docstring_flags.py [hunk 2]
```diff
@@ -82,7 +82,13 @@ class inline_flag(nodes.Inline, nodes.TextElement):
 
 def visit_inline_flag(self, node: inline_flag):
     flag_type = node.attributes["flag_type"]
-    html = f'<span class="flag {flag_type}">{flag_type}</span>'
+    html = f"""
+    <span class="flag {flag_type}">
+      <span class="hidden">(</span>
+      {flag_type}
+      <span class="hidden">)</span>
```
Comment (hellendag) [inline docs/sphinx/_ext/dagster-sphinx/dagster_sphinx/docstring_flags.py:89]: The hidden parens are picked up by Docs search scraping.

Comment (smackesey) [reply docs/sphinx/_ext/dagster-sphinx/dagster_sphinx/docstring_flags.py:89]: Mind adding this as a comment in the source

**Review by smackesey — APPROVED** [review]
Sweet.

One inline comment, and also my original PR added an orange color for experimental in the tailwind theme which you should be able to remove now.

**Pull request merged**